addons/l10n_in_ewaybill_stock/models/l10n_in_ewaybill.py

```diff
@@ -132,6 +133,34 @@ def action_set_to_challan(self):
             'state': 'challan',
         })
 
+    def action_print(self):
+        self.ensure_one()
+        if self.state == 'generated':
+            return super().action_print()
+        if self.state != 'challan':
+            raise UserError(_(
+                "Please generate the E-Waybill or mark the document as a Challan to print it."
+            ))
+
+        pdf_content = self.env['ir.actions.report']._render_qweb_pdf(
```

Comment:
Is not the code a bit duplicate with what we have already in l10n_in_ewaybill?  Maybe we could do a method in l10n_in_ewaybill that we can call from here as well?